### DISCUSSION TITLE
Restructure cache details menu

### DISCUSSION THREAD

**Discussion by moving-bits** [cgeo/cgeo#9692] [Q&A] [answered]
Currently the menu of a single cache (details page) is a long list, seemingly without any coherence:

- navigation symbol
- navigate to symbol
- log visit
- toggle talking
- add to calendar
- caches around
- show in browser
- store
- refresh
- remove
- remove user-defined waypoints
- vote
- share cache
- checker
- search challenge checker
- ignore
- extract waypoints
- export (with submenu for GPX, field notes, personal notes)
- personal note
- prevent waypoints from note
- set cache icon

(Some entries depend on cache type, member type etc.)

I have no ready solution to simplify or structure this cache, just wanted to start a discussion about it.

Some ideas:
- Remove the "store", "refresh" and "remove" entries, as we have buttons for it on the main cache page and the cache popup.
- Maybe "share" could be put into a button as well, left of the "reload" button.
- "Vote" should only be shown, if "GCVote" service is activated.
- "remove user-defined waypoints", "extract waypoints" and "prevent waypoints from note" could be put into a submenu "waypoints"

**Comment by Lineflyer** [comment]
You are right, I was also very surprised when I opened it first since longer time. Seems new items have been appended...

Regarding your ideas:
> Remove the "store", "refresh" and "remove" entries, as we have buttons for it on the main cache page and the cache popup.

That was also my first thought. Not sure, however how many users really use this way, but I guess most will use the icons.

> Maybe "share" could be put into a button as well, left of the "reload" button.

I am not a fan of moving a function into a totally different place and appearance.

> "Vote" should only be shown, if "GCVote" service is activated.

I agree. Isn't it like that.

> "remove user-defined waypoints", "extract waypoints" and "prevent waypoints from note" could be put into a submenu "waypoints"

I would remove "prevent waypoint from note" totally, as it is doubled to the personal note edit window. It belongs more to this place rather than in the caches main menu IMHO.
The other I would leave at their place for now.

Another topic while we are on it:
Would it be possible to add the menu visibility (if Room, with or without text, etc.) to each item in your above posting. I also saw a bit inconsistency in there as some of them are shown with text, some without (especially on tablet and/or landscape view).

**Reply by moving-bits** [reply]
> Another topic while we are on it:
> Would it be possible to add the menu visibility (if Room, with or without text, etc.) to each item in your above posting. I also saw a bit inconsistency in there as some of them are shown with text, some without (especially on tablet and/or landscape view).

What would be the expected target state?
Do we need to provide icons for all menu entries, then? (Something I would like to avoid)

Another inconsistency: Cache details menu disables some menu entries ("grey out"), whereas most other menus set non-fitting entries to "not visible". What would be the expected state for this?

**Comment by Lineflyer** [comment]
> What would be the expected target state?

e.g. I would like to skip the text on some of the items if they are visible as icon at the top bar.
One example: If I have my device in portrait mode I get two navigation icons shown, if I turn it into landscape I would expect to see 1-2 more entries there, but it stays same as in portrait. 

But you are right, that this is no urgent change. In the map menu I do see more potential to clean up (as on there in landscape mode I do get text with icon instead of more icons.

**Reply by moving-bits** [reply]
So you propose to clear the "withText" attribute for all menu items which have an icon associated?

**Comment by Lineflyer** [comment]
> Cache details menu disables some menu entries ("grey out"), whereas most other menus set non-fitting entries to "not visible". What would be the expected state for this?

Which are the potentially greyed out ones?
If there is no neccessity to see them, we should make them invisible.

**Reply by moving-bits** [reply]
Sorry, I was in "Cache list" menu, not "cache details". One example is having an empty list, where you can still see most of the menu entries, but mostly greyes out. IMHO there's no real need to show them.

**Reply by Lineflyer** [reply]
I remember a discussion about this topic and IMHO we ended with something like:
- If some menu items change depending on the object details, we should hide/show them as needed (e.g. Add to calendar only for events). The user is not expecting them, will not miss them and he cannot change it.
- If we have a more or less static menu (like for the cache list) we should instead grey out items which are not applicable due to some not fulfilled precondition controlled by the user. (e.g. cache list has no entries, move is greyed out). So the user is able to find the function and can ask himself what to do to enable it (because he can influence it).

Was this understandable?

**Reply by Lineflyer** [reply]
That would mean for cache details, e.g.:
- Open geochecker  -> Remove if no checker found
- Search challenge checker -> Remove if no challenge
- Vote -> Grey out as long as the cache is not found by user (this might be a future option as currently in GCVote you can also vote for not found caches...just an example here)

**Comment by Lineflyer** [comment]
> One example: If I have my device in portrait mode I get two navigation icons shown, if I turn it into landscape I would expect to see 1-2 more entries there, but it stays same as in portrait.

Forget this comment. First of all its no matching example, secondly I think I discovered a bug:
When I open a cache in portrait and turn my device the top bar stays same as in portrait.
When I open a cache in landscape I get more icons (as expected) but when then turning to portrait mode the icons stay same as in landscape and block the cache title out of the top bar.

I will transfer this into a dedicated issue.

**Comment by moving-bits** [comment, marked as answer]
Coming back to the menu structure, here's a refined proposal:

Starting point was: What function will probably needed frequently for the same cache while being on a tour. Those menu entries should stay on first menu level. So I came up with the following structure:

- navigation symbol
- navigate to symbol
- log visit _(Ok, it's used only once or twice per cache, but as this is a pretty important entry, it should stay top level, IMHO.)_
- toggle talking
- waypoints
  - remove user-defined waypoints
  - extract waypoints
  - prevent waypoints from note _(IMHO this was wished for as separate menu entry to have it available on waypoints page)_
- share
  - share cache
  - export GPX
  - export field notes _(or is it "upload field notes"?)_
  - export personal notes
- extra
  - add to calendar
  - show in browser
  - ignore
  - vote
  - caches around
  - set cache icon
- checker
- search challenge checker
- personal note

(Order of menu entries to be determined)

So it's mainly putting a couple of the less frequently menu entries into a new "extra" submenu, a couple of waypoint-related entries into a new "waypoint" submenu, and combine current "export" submenu with the "share" menu entry.

**Reply by Lineflyer** [reply]
Generally it looks fine to me, however some small adjustments I would suggest:

We should take care to reuse existing strings as much as possible as 1. new translations might take a time 2. users are familiar with current translations and retranslation might lead to a different result for the same function

You tend to write every menu item in lower case. Be aware that capitalization is also used in English at least for a the first letter and in case of things like headlines and menu items even IMHO each word except prepositions and conjunctions (I might be wrong, maybe someone is around with english as native language??).

> Share

I would rather name this new submenu "Share/Export" as it contains both functions then

> Share
>  - Export personal note

I always keep struggling with having this inside an export menu as exporting is something I get out of a tool (usually as a file), while that function in fact uploads the personal note to the server rather than exporting it.
Its anyway not really mandatory in a single cache menu as it is also available in the personal note section, so as a compromise we could put it into "Extra" and rename it to "Upload Personal Note" or simply remove it (=my favorite option).

> Extra

Good collection item, but I would rather name it "Advanced"

> Show in browser

Not sure if this should be buried in "Extra" as with c:geo you might need it rather frequent (e.g. for mysteries or if the listing contains color or table elements, which are not clearly visible). Rather keep it on main level?

> (Order of menu entries to be determined)

I would leave the order for all items remaining on top level as they currently are (to be reviewed after implementation) and put all new submenus below them (to be reviewed after implementation).

**Reply by moving-bits** [reply]
Thanks for reviewing, will work on it as time permits.

> You tend to write every menu item in lower case.

That was just my lazyness in writing down the menu entries. I would stay with today's names and capitalization (if no explicit change is proposed).

> > * Export personal note

I did not know the difference between "export => field notes" and "export => personal note" (and hadn't looked it up yet). But if "export => personal note" really is just the same as the button below the PN field, then we should omit this menu entry.

> Good collection item, but I would rather name it "Advanced"

Ok

> > Show in browser
> 
> Not sure if this should be buried in "Extra" as with c:geo you might need it rather frequent (e.g. for mysteries or if the listing contains color or table elements, which are not clearly visible). Rather keep it on main level?

Let's try it on main level, and see how the results look in total.

> I would leave the order for all items remaining on top level as they currently are (to be reviewed after implementation) and put all new submenus below them (to be reviewed after implementation).

Ok

**Reply by moving-bits** [reply]
see #9815 for implementation

**Comment by jcklpe** [comment]
I'd be willing to help out with this as a professional UX designer, if there was some interest in collaboration etc.

**Reply by moving-bits** [reply]
Thanks, but the changes proposed here regarding the menu structure are already incorporated on `master` branch, so the original issue/discussion should have been closed already. - Having said that: If you find more need to change things in the menus, feel free to open a dedicated issue for that.
